### PR TITLE
Skip auditd and sysctl tests for containers

### PR DESCRIPTION
**Pull request by artem-sidorenko**
See https://github.com/dev-sec/chef-os-hardening/pull/199 for reference

Signed-off-by: Artem Sidorenko <artem@posteo.de>

### CODE REVIEW

File: controls/os_spec.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
 shadow_group = 'root'
 shadow_group = 'shadow' if os.debian? || os.suse?
+container_execution = virtualization.role == 'guest' && virtualization.system =~ /^(lxc|docker)$/
```
Comment (chris-rock) [inline controls/os_spec.rb]: In a next iteration, we should probably capture this logic in our custom inspec resource

**Comment by artem-sidorenko** [comment]
@chris-rock @atomic111 thank you!

@atomic111 done

**Comment by artem-sidorenko** [comment]
@chris-rock may I ask you to have a look here at travis? The tests are running fine, inspec check throws an error. It looks the data strucutre in `virtualization.rb` isn't properly initialized when running within `inspec check`.  I can reproduce it locally on the latest inspec version too:

```
NoMethodError: undefined method `[]' for nil:NilClass
  /home/travis/build/dev-sec/linux-baseline/vendor/bundle/ruby/2.4.0/gems/inspec-1.51.21/lib/resources/virtualization.rb:39:in `block (2 levels) in <class:Virtualization>'
  /home/travis/build/dev-sec/linux-baseline/controls/os_spec.rb:30:in `load_with_context'
...
```

Ideas?

**Comment by chris-rock** [comment]
We can circumvent that by using a proper InSpec resource for that. That keeps the control file clean. I cannot start to implement this before next Monday :-(

**Comment by jquick** [comment]
inspec check should be working tomorrow when we release the new version.

**Comment by artem-sidorenko** [comment]
@jquick cool, thanks :-)

**Pull request merged**